libappstream-glib/as-store.c

```diff
@@ -2269,6 +2326,10 @@ as_store_from_bytes (AsStore *store,
 		return as_store_from_xml (store, tmp, NULL, error);
 	}
 
+	/* is a DEP-11 file */
+	if (g_strcmp0 (content_type, "text/plain") == 0 && is_dep11_data (bytes))
```

Comment:
Any idea why yaml files are detected as `text/plain`?

Reply:
I don't think it's possible to reliably detect YAML as there's nothing consistently used in the start of a YAML file that would reliably indicate that. Even the `is_dep11_data` function is pretty vague in checking, though it shouldn't collide with any other appstream file format and will fail if it turns out not to be YAML.